src/main/java/frc/robot/subsystems/vision/PhotonSubsystem.java

```diff
@@ -99,18 +100,18 @@ private double getTime() {
     return lastPose.timestampSeconds;
   }
 
-  private boolean checkPoseValidity(EstimatedRobotPose pose) {
-    double distanceBetweenPoses =
+  public boolean checkPoseValidity(
+      Pose3d checkPose, Pose3d confirmedPose, double checkTimestamp, double confirmedTimestamp) {
+    final double distanceBetweenPoses =
         Math.sqrt(
-            Math.pow(pose.estimatedPose.getX() - lastPose.estimatedPose.getX(), 2)
-                + Math.pow(pose.estimatedPose.getY() - lastPose.estimatedPose.getY(), 2)
+            Math.pow(checkPose.getX() - confirmedPose.getX(), 2)
+                + Math.pow(checkPose.getY() - confirmedPose.getY(), 2)
                 + Math.pow(
-                    pose.estimatedPose.getZ() - lastPose.estimatedPose.getZ(),
+                    checkPose.getZ() - confirmedPose.getZ(),
                     2)); // 3D distance formula (same one as used in periodic)
     if ((distanceBetweenPoses > Constants.VisionConstants.MAX_ACCEPTABLE_ERROR_METERS)
-        && (Math.abs(pose.timestampSeconds - lastPose.timestampSeconds)
+        && (Math.abs(checkTimestamp - confirmedTimestamp)
```

Comment:
If this error happens, won't we no longer be able to accept new poses? Since we don't change the lastPose, and any future poses will have a later timestamp.